raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -649,81 +652,172 @@ function receiveTransferExpired(
+ *
+ * @param state$ - Observable of current state
+ * @param action - Withdraw request which caused this handling
+ * @param deps - RaidenEpicDeps members
+ * @param deps.address - Our address
+ * @param deps.log - Logger instance
+ * @param deps.signer - Signer instance
+ * @param deps.network - Current Network
+ * @param deps.config$ - Config observable
+ * @returns Observable of withdrawMessage.request|withdraw.failure actions
+ */
+function sendWithdrawRequest(
+  state$: Observable<RaidenState>,
+  action: withdraw.request,
+  { log, address, signer, network, config$ }: RaidenEpicDeps,
+): Observable<withdrawMessage.request | withdraw.failure> {
+  if (action.meta.direction !== Direction.SENT) return EMPTY;
+  return combineLatest([state$, config$]).pipe(
+    first(),
+    mergeMap(([state, { revealTimeout }]) => {
+      const channel = state.channels[channelKey(action.meta)];
+      assert(channel?.state === ChannelState.open, 'channel not open');
+      if (
+        channel.own.pendingWithdraws.some(
+          (req) =>
+            req.total_withdraw.eq(action.meta.totalWithdraw) &&
+            req.expiration.eq(action.meta.expiration),
+        )
+      )
+        return EMPTY; // already requested, skip but don't fail
+      assert(
+        action.meta.expiration >= state.blockNumber + revealTimeout,
+        'expired or expires too soon',
+      );
+      assert(
+        action.meta.totalWithdraw.gt(channel.own.withdraw),
+        'withdraw lower than previous totalWithdraw',
+      );
+      assert(
+        action.meta.totalWithdraw.lte(channelAmounts(channel).ownTotalWithdrawable),
+        'not enough capacity',
+      );
+      const request: WithdrawRequest = {
+        type: MessageType.WITHDRAW_REQUEST,
+        message_identifier: makeMessageId(),
+        chain_id: bigNumberify(network.chainId) as UInt<32>,
+        token_network_address: action.meta.tokenNetwork,
+        channel_identifier: bigNumberify(channel.id) as UInt<32>,
+        participant: address,
+        total_withdraw: action.meta.totalWithdraw,
         nonce: channel.own.nextNonce,
-        expiration: request.expiration,
+        expiration: bigNumberify(action.meta.expiration) as UInt<32>,
       };
-      confirmation$ = from(signMessage(signer, confirmation, { log })).pipe(
-        tap((signed) => cache.put(cacheKey, signed)),
+      return from(signMessage(signer, request, { log })).pipe(
+        map((message) => withdrawMessage.request({ message }, action.meta)),
       );
-    }
-  }
+    }),
+    catchError((err) => of(withdraw.failure(err, action.meta))),
+  );
+}
 
+/**
+ * Validates a received WithdrawConfirmation message
+ *
+ * @param state$ - Observable of current state
+ * @param action - Withdraw request which caused this handling
+ * @param deps - RaidenEpicDeps members
+ * @param deps.address - Our address
+ * @returns Observable of withdrawMessage.success actions
+ */
+function receiveWithdrawConfirmation(
```

Comment:
This is a little bit of an exception to the rule: no function called by `transferGenerateAndSignEnvelopeMessageEpic` receives the `$` suffix, because generally they're split in the top-level and main `mergeMap`'s callback, which then receives the `$`, and both are named similarly, just differing in the `$` suffix. Since on this case the function was small enough to not split, the only `receiveWithdrawConfirmation` didn't receive the `$` suffix to match their friends.